core/src/main/java/org/hibernate/ogm/loader/impl/OgmLoader.java

```diff
@@ -488,10 +551,45 @@ private TupleAsMapResultSet getResultSet(Serializable id, OgmLoadingContext ogmL
 		final TupleAsMapResultSet resultset = new TupleAsMapResultSet();
 		if ( getEntityPersisters().length > 0 ) {
 			OgmEntityPersister persister = getEntityPersisters()[0];
-			final EntityKey key = EntityKeyBuilder.fromPersister( persister, id, session );
-			Tuple entry = gridDialect.getTuple( key, persister.getTupleContext() );
-			if ( entry != null ) {
-				resultset.addTuple( entry );
+			if ( useBatching( qp ) ) {
+				// here we expect to receive QueryParameters.positionalParameters full of ids and thus of the same type.
+				// if that's not the case, we are in a bit of a trouble :)
+				// TODO Adding a safeguard but we pay the price for each call, should we remove it?
```

Comment:
In which case the parameter type would be a wrong type? Seems like its a programming error which we should detect during testing/development, so it's not needed to be asserted continuously at runtime?